watch.mjs

```diff
@@ -0,0 +1,42 @@
+import fsCallback from 'node:fs';
+import { buildComponentSets, listComponentSetPaths, validateComponentSet } from './build.mjs';
+
+export async function watchComponentSets() {
+    // Watch does not rebuild vendor.js and design.css
+    await buildComponentSets();
+
+    for (const componentSetPath of await listComponentSetPaths()) {
+        await watchComponentSet(componentSetPath);
+    }
+}
+
+export async function watchComponentSet(componentSetPath) {
+    let dirty = false;
+    let activePromise = null;
+
+    const doValidateComponentSet = () => {
+        dirty = false;
+        activePromise = validateComponentSet(componentSetPath)
+            .catch((err) => {
+                console.error(err);
+            })
+            .finally(() => {
+                activePromise = undefined;
+
+                if (dirty) {
+                    doValidateComponentSet();
+                }
+            });
+    };
+
+    fsCallback.watch(componentSetPath, { recursive: true }, (eventType, filename) => {
```

Comment:
See https://nodejs.org/docs/latest/api/fs.html#caveats. This option only works on MacOS and Windows, but I believe it is good enough for the use case.